### PR TITLE
Upgrade the Dockerfile to Debian buster

### PR DESCRIPTION
**Pull request by tranchitella**
We require libssl1.1 >= 1.1.1 to enable ECC keys in the Mender Client.
This version of openssl is not available in stretch, thus we upgrade to
buster.

Changelog: none

Signed-off-by: Fabio Tranchitella <fabio.tranchitella@northern.tech>

Requires: https://github.com/mendersoftware/mender-test-containers/pull/72

### CODE REVIEW

**Review by mzedel — APPROVED** [review]

**Review by alfrunes — APPROVED** [review]

File: Dockerfile [hunk 1]
```diff
@@ -1,4 +1,4 @@
-ARG DEBIAN_VERSION=stretch
+ARG DEBIAN_VERSION=buster
 FROM debian:$DEBIAN_VERSION
```
Comment (mzedel) [inline Dockerfile:3]:
```suggestion
ARG DEBIAN_VERSION=buster
FROM debian:$DEBIAN_VERSION
ARG DEBIAN_VERSION
```

based on [this](https://docs.docker.com/engine/reference/builder/#understand-how-arg-and-from-interact)

**Comment by lluiscampos** [comment]
We have been deliberately using Debian `stretch` and the older toolchain with ARMv6 to support a wider range of OS and devices, namely Raspberry Pi Zero W. See also this [old discussion](https://github.com/mendersoftware/mender-dist-packages/pull/13#discussion_r328451238).

At the time was only liblzma, today is also libssl and libglib2.0 (for D-Bus) so it might be time to revisit this decision.

I need to think a bit more to form a proper opinion. I'll come back to this later today

**Comment by kacf** [comment]
Following the code that @alfrunes posted in the chat:

```c
#if OPENSSL_VERSION_NUMBER >= 0x1010100fL
...

int X_EVP_DigestSignInit(EVP_MD_CTX *ctx, EVP_PKEY_CTX **pctx,
		const EVP_MD *type, ENGINE *e, EVP_PKEY *pkey){
	return EVP_DigestSignInit(ctx, pctx, type, e, pkey);
}
...
#else
...
int X_EVP_DigestSignInit(EVP_MD_CTX *ctx, EVP_PKEY_CTX **pctx,
		const EVP_MD *type, ENGINE *e, EVP_PKEY *pkey){
	return 0;
}
...
#endif
```

We could perhaps use `dlopen("EVP_DigestSignInit")` instead?

**Comment by lluiscampos** [comment]
I think we should keep the criteria of building with the previous-to-stable release. However, [next Debian release is coming this month](https://en.wikipedia.org/wiki/Debian_version_history#Debian_11_(Bullseye)), so we can update this soon. But we need a JIRA task to, among other things, test the updated packages on Raspberry Pi Zero W.

I think @kacf suggestion is good regardless: it will make the software in effect check the openSSL version installed in runtime, not at compile time.

**Comment by lluiscampos** [comment]
I think that you think that by updating the submodule you are testing with the new raspbian testing image, but that is not true. The tests use the image from upstream (from Docker Hub). So that part needs manual testing at least once.

**Comment by tranchitella** [comment]
@lluiscampos I'm fully aware updating the submodule doesn't influence the test, and I launched this PR for discussing with you who are experts in these topics.

**Comment by lluiscampos** [comment]
@tranchitella Okey, sorry.

I'll run the tests locally.

**Comment by tranchitella** [comment]
@kacf can we merge this?

**Comment by alfrunes** [comment]
> Following the code that @alfrunes posted in the chat:
> 
> ```c
> #if OPENSSL_VERSION_NUMBER >= 0x1010100fL
> ...
> 
> int X_EVP_DigestSignInit(EVP_MD_CTX *ctx, EVP_PKEY_CTX **pctx,
> 		const EVP_MD *type, ENGINE *e, EVP_PKEY *pkey){
> 	return EVP_DigestSignInit(ctx, pctx, type, e, pkey);
> }
> ...
> #else
> ...
> int X_EVP_DigestSignInit(EVP_MD_CTX *ctx, EVP_PKEY_CTX **pctx,
> 		const EVP_MD *type, ENGINE *e, EVP_PKEY *pkey){
> 	return 0;
> }
> ...
> #endif
> ```
> 
> We could perhaps use `dlopen("EVP_DigestSignInit")` instead?

I think that would be tricky as the respective `EVP_` symbols [exist](https://www.openssl.org/docs/man1.0.2/man3/EVP_DigestSignInit.html) for `1.0` as well. Maybe there's some other symbol we could check for?
Anyways, I think we can merge this PR as it's somewhat unrelated to the above issue.

**Comment by kacf** [comment]
Yeah, we probably don't need that fix if we're building for a release that have all of them anyway. I'm ok with merging as is.
